File: Week-1-JS-Recap/problems.js
// Reverse a string
// Find the max number in an array
// Determine if a word is a palindrome

// Problem 1
function reverseStr(str) {
    let resultStr = ""
    for (let char of str) {
        resultStr = char + resultStr
    }

    return resultStr
}

console.log(reverseStr("word"))

// problem 2
function findLargestNum(nums) {
    let resultNum = nums[0]

    for (let num of nums) {
        if (num > resultNum) {
            resultNum = num
        }
    }

    return resultNum
}

console.log(findLargestNum([-3, 10, 3, 4]))

// Problem 3
function isPalindrome(str) {
    var tempStr = str
    for (let char of str) {
        currChar = tempStr.slice(-1)
        tempStr = tempStr.slice(0, -1)
        if (char !== currChar) {
            return false
        }
    }
    return true
}

console.log(isPalindrome("car"))
console.log(isPalindrome("tots"))
console.log(isPalindrome("racecar"))

exports.reverseStr = reverseStr
exports.findLargestNum = findLargestNum
exports.isPalindrome = isPalindrome

File: Week-2-Python/problems.py
def reverse_string(s):
    res = ""
    for char in s:
        res = char + res
    return res

print(reverse_string("boo"))


def find_max(list):
    result_num = list[0]

    for num in list:
        if num > result_num:
            result_num = num
    
    return result_num

print(find_max([0, 2, 4, 2]))


def is_palindrome(s):
    temp_str = s 
    res = ""
    for char in s:
        curr_char = temp_str[len(temp_str) - 1]
        temp_str = temp_str[:-1]
        if char != curr_char:
            return False

    return True

print(is_palindrome("racecar"))
print(is_palindrome("treat"))